client/internal/httpsender.go

```diff
@@ -167,6 +168,9 @@ func (h *HTTPSender) sendRequestWithRetries(ctx context.Context) (*http.Response
 						interval = recalculateInterval(interval, resp)
 						err = fmt.Errorf("server response code=%d", resp.StatusCode)
 
+					case http.StatusNotFound:
```

Comment:
Looks at this again, I am not sure I understand why this case is needed. We have a `default` later which returns an error for all other cases including StatusNotFound. Can you clarify what the purpose of this separate case is?

Reply:
@tigrannajaryan  got it. So, I removed the extra case and test cases and only kept `missing logging statement` which returns if any error is found. This sounds good?